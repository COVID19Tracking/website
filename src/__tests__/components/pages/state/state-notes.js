import React from 'react'
import renderer from 'react-test-renderer'
import StateNotes from '~components/pages/state/state-notes'

const sampleText = `This is the first paragraph for June 13.

This second line tests years, as in July 14, 2020.

You can also format with August 18 2020, I guess that is fine too.`

describe('Components : Pages : State : State notes', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StateNotes notes={sampleText} stateName="California" />)
      .toJSON()
    expect(tree).toMatchSnapshot()

    const treeHiddenLede = renderer
      .create(<StateNotes notes={sampleText} hideLede />)
      .toJSON()
    expect(treeHiddenLede).toMatchSnapshot()
  })
})
