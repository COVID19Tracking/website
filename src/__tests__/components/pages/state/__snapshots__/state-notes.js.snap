// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : State : State notes renders correctly 1`] = `
<div
  className="container container container"
>
  <div
    className="full"
  >
    <div
      className="notes-expandable expandable"
    >
      <span
        className="label"
      >
        Notes: 
      </span>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>This is the first paragraph for <strong>June 13</strong>.</p>
",
          }
        }
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>This second line tests years, as in <strong>July 14, 2020</strong>.</p>
",
          }
        }
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "",
          }
        }
      />
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<p>You can also format with <strong>August 18 2020</strong>, I guess that is fine too.</p>
",
          }
        }
      />
      <div
        className="fader"
      />
    </div>
    <button
      aria-hidden={true}
      className="expand"
      onClick={[Function]}
      type="button"
    >
      Read more
       
      <span
        className="arrow"
      >
        â†“
      </span>
    </button>
  </div>
</div>
`;
